design-toolkit/components/src/components/date-field/index.tsx

```diff
@@ -155,51 +108,27 @@ const DateInput = ({
     <div className='relative flex'>
       {size === 'medium' ? (
         <Icon
-          className={cn([
-            '-translate-y-1/2 absolute top-1/2 left-s',
-            props.isDisabled ? 'text-disabled' : 'text-default-light',
-          ])}
+          className={`-translate-y-1/2 absolute top-1/2 left-s ${props.isDisabled ? ' text-disabled' : ' text-default-light'}`}
         >
           <Calendar />
         </Icon>
       ) : null}
       <AriaDateInput
         {...props}
-        className={({ isDisabled, isInvalid }) =>
-          cn(
-            dateFieldStyles({
-              isDisabled,
-              isInvalid,
-              isReadOnly: isReadOnly,
-              size,
-              className,
-            }),
-          )
-        }
+        className={composeRenderProps(className, (className) =>
+          DateFieldStyles({
+            isDisabled: props.isDisabled,
+            isInvalid: props.isInvalid,
+            isReadOnly,
+            size,
```

Comment:
You can set the size to a data attribute: https://github.com/gohypergiant/standard-toolkit/blob/main/design-toolkit/components/src/components/button/index.tsx#L69 and this makes it available to use as a variant selector: https://github.com/gohypergiant/standard-toolkit/blob/main/design-toolkit/components/src/variants/variants.css#L17